src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -80,7 +85,11 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             // No AT in the cache or AT needs to be refreshed
             try
             {
-                return await FetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                using (logger.LogBlockDuration("[OBO request] Fetching OBO token from ESTS"))
+                {
+                    var result = await FetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Can you simply return without using the result?

Reply:
I'm going to leave it in. it helps when debugging. the compiler will optimize it anyway, there's not perf impact.